pkg/snapshot/restorer/restorer_test.go

```diff
@@ -71,116 +79,522 @@ var _ = Describe("Running Restorer", func() {
+				restoreOptions.BaseSnapshot.SnapName = ""
+				err := rstr.Restore(restoreOptions)
+				Expect(err).Should(BeNil())
+
+			})
+		})
+
+		Context("with invalid snapdir and snapname", func() {
+			fmt.Println("Testing for invalid snapdir and snapname")
+			It("should fail to restore", func() {
+
+				restoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				restoreOptions.BaseSnapshot.SnapDir = "test"
+				restoreOptions.BaseSnapshot.SnapName = "test"
+				err := rstr.Restore(restoreOptions)
+				Expect(err).Should(HaveOccurred())
+
+			})
+		})
+
+		Context("with maximum of one fetcher allowed", func() {
+			fmt.Println("Testing for max-fetchers=1")
+			It("should restore etcd data directory", func() {
+				maxFetchers = 1
+
+				restoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				err = rstr.Restore(restoreOptions)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				err = checkDataConsistency(restoreDataDir, logger)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+		})
+
+		Context("with maximum of four fetchers allowed", func() {
+			fmt.Println("Testing for max-fetchers=4")
+			It("should restore etcd data directory", func() {
+				maxFetchers = 4
+
+				restoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				err = rstr.Restore(restoreOptions)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				err = checkDataConsistency(restoreDataDir, logger)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+		})
+
+		Context("with maximum of hundred fetchers allowed", func() {
+			fmt.Println("Testing for max-fetchers=100")
+			It("should restore etcd data directory", func() {
+				maxFetchers = 100
```

Comment:
Do we have 100 delta snapshots to be restored?

Reply:
the default duration to run snapshotter is 100s but when but as the delta snapshot period is 5s so in total this creates around 21 snapshots. we can change the deltaSnapshotPeriod to 1s and we will have 100 delta snapshots to apply. @shreyas-s-rao  -- wdyt?

Reply:
I've now made detlaSnapshotDuration to 20s as default with  deltaSnapshotPeriod  now set to 1s.